### PR TITLE
Fix iterator for immutable values from imported types

### LINKED ISSUE #30566: Map iterator does not work for map<record> configurable variables from non-default module
**Description:**
When there is a `map<record>` typed configurable variable created from an imported map type of a non-default module, 
the map iterator function gives a no record found error. 

**Steps to reproduce:**
In a package where default module name is `main` and a non-default module with name `main.mod1`,
In mod1.bal,
```ballerina
public type Student record {
    string name = "";
    int id = 1;
};
public type StudentMap map<Student>;
```
In main.bal,
```ballerina
import ballerina/io;
import main.mod1;
configurable mod1:StudentMap & readonly studentMap = ?;

public function main() {
    int count = 0;
    foreach var entry in studentMap {
        count += 1;
    }
    io:println(count);
}
```
And in `Config.toml`,
```toml
[studentMap]
student1.id = 11
student1.name = "Anne"
student2.id = 22
student2.name = "Bob"
student3.id = 33
student3.name = "Charles"
```
Error:
```
error: No such record: Student & readonly
        at ballerina.lang.map.1_1_0:externNext(internal.bal:35)
           ballerina.lang.map.1_1_0.MapIterator:next(internal.bal:31)
```
**Affected Versions:**
beta1

**OS, DB, other environment details and versions:**

**Related Issues (optional):**
#30389 

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
Fix the map/table iteration failing with `No such record` error when we iterate over a map/table of records that is imported from another module
Fixes #30566 

## Approach
The failure is due to the iterator call the getZeroValue() of the value from a  wrong package. 
Fixed it by providing the empty value  from the original package for `getZeroValue()` call.

## Samples

In main.mod1.bal
```ballerina
public type Student record {
    string name = "";
    int id = 1;
};
public type StudentMap map<Student>;
```
In main.bal,
```ballerina
import main.mod1;
import ballerina/io;

public function main() {
    mod1:StudentMap studentMap = {a : {name: "aaa"}, b : {name: "bbb"}};
    int count = 0;
    foreach var entry in studentMap.cloneReadOnly() {
        count += 1;
    }
    io:println(count);
}
```

## Remarks
Related  #32283

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-integration-test/src/test/resources/configurables/configStructuredTypesProject/configOpenRecord/main.bal [hunk 1]
```diff
@@ -37,8 +37,23 @@ function testOpenRecords() {
...
+
     testRecordCollection();
+    testArrayIteration();
+    testTableIteration();
+
     rest_fields:testRestFields();
     util:print("Tests passed");
 }
+
+
```
Comment (Nadeeshan96) [inline tests/jballerina-integration-test/src/test/resources/configurables/configStructuredTypesProject/configOpenRecord/main.bal:53]: There is an extra newline here if it is not by intention.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BRecordType.java [hunk 2]
```diff
@@ -111,7 +111,11 @@ public BRecordType(String typeName, Module pkg, long flags, Map<String, Field> f
 
     @Override
     public <V extends Object> V getZeroValue() {
-        return (V) ValueCreator.createRecordValue(this.pkg, this.typeName);
+        String typeName = this.typeName;
+        if (intersectionType != null) {
+            typeName = ((BIntersectionType) intersectionType).getConstituentTypes().get(0).getName();
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BRecordType.java:116]: @MaryamZi Is there any other way we can get the mutable type?

**Comment by warunalakshitha** [comment]
Lets send this to bets 3 branch

**Comment by HindujaB** [comment]
> Lets send this to bets 3 branch

Moved the base branch here :
https://github.com/ballerina-platform/ballerina-lang/pull/32444

**Pull request closed without merging**